shtdlib.sh

```diff
@@ -773,23 +807,24 @@ function set_dir_perm_owner {
 # but could be an issue if used in a different way. Third and fourth parameters
 # are optional
 function copy_file {
-    rsync_base_flags="-ltDu  --inplace --backup --backup-dir=\"${backup_dir}\" --keep-dirlinks"
+    rsync_base_flags="-ltDu  --inplace --backup --backup-dir=\"${backup_dir:-${dest}.backup}\" --keep-dirlinks"
     local source="${1}"
     local dest="${2}"
     local owner_group="${3:-}"
     local perm="${4:-}"
     set -f
-    find_directory="$(dirname ${source})"
-    find_pattern="$(basename ${source})"
+    find_directory="$(dirname "${source}")"
+    find_pattern="$(basename "${source}")"
     set +f
     debug 10 "Called copy_file with ${source} ${dest} ${owner_group} ${perm}"
+    # shellcheck disable=SC2086
     if [ -e "${source}" ] ; then
         debug 10 "Filesystem object ${source} exists"
         # Make sure permissions and owner are OK
         set_file_perm_owner "${source}" "${owner_group}" "${perm}"
         if [ -f "${source}" ] ; then
             debug 10 "Found file ${source}"
-            if ${force_overwrite} ; then
+            if ${force_overwrite:-false} ; then
```

Comment:
Does this require quotes too?

Reply:
I don't think so, since true/false are actual commands and if uses their return status (0 or 1) adding quotes would make if try to interpret the string right ?